app/forms/AssistanceDetailsForm.scala

```diff
@@ -94,20 +100,26 @@ object AssistanceDetailsForm {
           case true => "Yes"
           case false => "No"
         },
-        ad.needsSupportForOnlineAssessment match {
-          case true => "Yes"
-          case false => "No"
-        },
+        toOptString(ad.needsSupportForOnlineAssessment),
         ad.needsSupportForOnlineAssessmentDescription,
-        ad.needsSupportAtVenue match {
-          case Some(true) => Some("Yes")
-          case Some(false) => Some("No")
-          case _ => None
-        },
-        ad.needsSupportAtVenueDescription
+        toOptString(ad.needsSupportAtVenue),
+        ad.needsSupportAtVenueDescription,
+        toOptString(ad.needsSupportForPhoneInterview),
+        ad.needsSupportForPhoneInterviewDescription
       )
     }
 
+    def toOptBoolean(optString: Option[String]) = optString match {
+      case Some("Yes") => Some(true)
+      case Some("No") => Some(false)
+      case _ => None
+    }
+
+    def toOptString(optBoolean: Option[Boolean]) = optBoolean match {
```

Comment:
In this case also the shorter:
optBoolean flatMap {
  case true => Some("Yes")
  case false => Some("No")
}

Should work but up to you.